.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDurationParser.cs

```diff
@@ -208,33 +207,34 @@ private DateTimeResolutionResult ParseNumberCombinedUnit(string text)
         private DateTimeResolutionResult ParseAnUnit(string text)
         {
             var ret = new DateTimeResolutionResult();
-            double numVal = 0;
-            string numStr, unitStr;
             var suffixStr = text;
-            Match match;
 
-            match = this.config.AnUnitRegex.Match(text);
+            var match = this.config.AnUnitRegex.Match(text);
             if (!match.Success)
             {
                 match = this.config.HalfDateUnitRegex.Match(text);
             }
 
             if (match.Success)
             {
-                numVal = match.Groups["half"].Success ? 0.5 : 1;
+                var numVal = match.Groups["half"].Success ? 0.5 : 1;
                 numVal += ParseNumberWithUnitAndSuffix(suffixStr);
-                numStr = numVal.ToString(CultureInfo.InvariantCulture);
+                var numStr = numVal.ToString(CultureInfo.InvariantCulture);
 
                 var srcUnit = match.Groups["unit"].Value.ToLower();
                 if (this.config.UnitMap.ContainsKey(srcUnit))
                 {
-                    unitStr = this.config.UnitMap[srcUnit];
+                    var unitStr = this.config.UnitMap[srcUnit];
 
-                    ret.Timex = "P" + (IsLessThanDay(unitStr) ? "T" : "") + numStr + unitStr[0];
+                    ret.Timex = TimexUtility.GenerateDurationTimex(numVal, unitStr, IsLessThanDay(unitStr));
                     ret.FutureValue = ret.PastValue = double.Parse(numStr) * this.config.UnitValueMap[srcUnit];
                     ret.Success = true;
-
-                    return ret;
+                }
+                else if (match.Groups["business"].Success)
+                {
+                    ret.Timex = TimexUtility.GenerateDurationTimex(numVal, "BD", false);
```

Comment:
We should not pass BD as a string into it. Parameters should define the input needed to generate the appropriate regex. The actual substring to use is an internal implementation detail of the Gen method.